ingestion_server/ingestion_server/cleanup.py

```diff
@@ -300,6 +302,43 @@ def save_cleaned_data(result: dict) -> dict[str, int]:
     return cleanup_counts
 
 
+def _upload_to_s3(fields):
+    """
+    Upload cleaned data to S3. It assumes that the bucket already exists.
+
+    Locally, it connects to a MinIO instance through its endpoint and test credentials.
+    On live environments, the connection is allowed via IAM roles.
+    """
+    bucket_name = config("OPENVERSE_BUCKET", default="openverse-catalog")
+    s3_path = "shared/data-refresh-cleaned-data"
+    try:
+        s3 = boto3.resource(
+            "s3",
+            endpoint_url=config("AWS_S3_ENDPOINT", default=None),
+            aws_access_key_id=config("AWS_ACCESS_KEY_ID", default=None),
+            aws_secret_access_key=config("AWS_SECRET_ACCESS_KEY", default=None),
+            region_name=config("AWS_REGION", default=None),
+        )
+        s3.meta.client.head_bucket(Bucket=bucket_name)
+        bucket = s3.Bucket(bucket_name)
+        log.info(f"Connected to S3 and '{bucket_name}' bucket loaded.")
+    except Exception as e:
+        log.error(f"Files upload failed. Error connecting to S3.\n{e}")
+        return
+
+    for field in fields:
+        file_path = TMP_DIR.joinpath(f"{field}.tsv")
+        if not file_path.exists():
+            continue
```

Comment:
Can you add a comment with the explanation you shared in the other PR for why this is an expected and good condition (not an error)? Otherwise, it's definitely not clear.

Reply:
> I don't understand the bit about how these files will over time no longer exist. Why would that happen? Once we've applied the fix upstream, is there a point where the data refresh would have "half fixed" data, and some of the cleaning would stop? 

The cleaning steps will remain for a time, let's say two or one data refresh process, until we're confident that nothing is left pending in that ETL. The cleaning won't stop midway by itself, so we'd need to catch these cases when the files aren't produced anymore.

Thanks for raising the flag here. It was clear to me, but now I see I was assuming many things. I added a comment that hopes to add more context to it.